pkg/tasks/introspect.go

```diff
@@ -0,0 +1,44 @@
+package tasks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/content-services/content-sources-backend/pkg/external_repos"
+	"github.com/content-services/content-sources-backend/pkg/tasks/queue"
+	"github.com/go-playground/validator/v10"
+	"github.com/rs/zerolog/log"
+)
+
+const Introspect = "introspect"
+
+type IntrospectPayload struct {
+	Url string
+}
+
+// TODO possibly remove context arg
+func IntrospectHandler(ctx context.Context, task *queue.TaskInfo) error {
```

Comment:
we might need to discuss a bit about how we'll handle the request-id header to know if we can remove the context or not, something we haven't discussed before

Reply:
I'll resolve this for now, and open a task to cover it